src/core/reactor.cc

```diff
@@ -4333,6 +4516,17 @@ void set_idle_cpu_handler(idle_cpu_handler&& handler) {
     engine().set_idle_cpu_handler(std::move(handler));
 }
 
+namespace experimental {
+future<process> spawn_process(std::filesystem::path pathname,
+                              spawn_parameters params) {
+    if (params.argv.empty()) {
+        return process::spawn(pathname);
+    } else {
+        return process::spawn(pathname, std::move(params));
```

Comment:
Don't understand this if. Why not always pass params? They could include more than the environment.

Reply:
ahh, indeed. i am exposing two APIs in `seastar.hh` now. one with params, one without.